### DISCUSSION TITLE
"Missing or insufficient permissions"

### DISCUSSION THREAD

**Discussion by akauppi** [firebase/firebase-js-sdk#4567] [Archive] [answered]
*Disclaimer: This *may* be a bug in my end; I'm not sure. Posting to see if others have experienced the same.*

![image](https://user-images.githubusercontent.com/1297734/109618985-e381f380-7b40-11eb-9bcb-3c9fc4a1e0f8.png)

I started getting these when transitioning the code to `@exp` API - this one is from `setDoc`. The access rights tests I have at the backend pass, and there has been no changes to the rules, since things worked with 8.x.

I'll disable the code now, since it's not essential.

**Comment by akauppi** [comment]
Also getting permission errors listening to data.

<img width="749" alt="image" src="https://user-images.githubusercontent.com/1297734/109667361-bc471880-7b78-11eb-911a-eff7da1467fc.png">

Maybe something is systematically wrong with my setup. Will check everything I can.

**Reply by akauppi** [reply]
I'm catching up some weird behaviour on the `currentUser` value. Is that connected to Firestore's access rights, or are they separate?

**Reply by Feiyang1** [reply]
Firestore does read from `currentUser` and send the corresponding credential to the backend for access check.  Sounds like it might be related to another [issue](https://github.com/firebase/firebase-js-sdk/discussions/4570) you reported, where `currentUser` becomes `null` unexpectedly.

@samhorlbeck @schmidt-sebastian Can you please take a look at it?

**Comment by akauppi** [comment, marked as answer]
Some changes caused Firestore auth to start working, so I think this was a one-timer on my end.
